src/xdocs/property_types.xml

```diff
@@ -831,16 +831,62 @@
 
     <section name="SeverityLevel">
       <p>
-        This type represents the severity level of a check violation. The
-        valid options are:
+        This type represents the severity level of a module violation.
+        The following table describes the valid options:
       </p>
 
-      <ul>
-        <li><code>ignore</code></li>
-        <li><code>info</code></li>
-        <li><code>warning</code></li>
-        <li><code>error</code></li>
-      </ul>
+      <div class="wrapper">
+        <table>
+          <tr>
+            <td>Option</td>
+            <td>Definition</td>
+          </tr>
+
+          <tr>
+            <td><code>ignore</code></td>
+            <td>
+              If a violation occurs, modules with this severity are ignored as if they never
+              occurred.
+              Checkstyle Audit Listeners will not display the violation and will not fail the
+              execution.
+              This severity can be used to disable a module in the configuration. However,
+              depending on the setting of
+              <a href="cmdline.html#Command_line_usage"><code>executeIgnoredModules</code></a>
+              when executed, the module may still run and if an exception occurs, the exception
+              will turn into an error and act as one.
+            </td>
+          </tr>
+
+          <tr>
+            <td><code>info</code></td>
+            <td>
+              If a violation occurs, modules with this severity are displayed as informational.
+              Checkstyle will display the violation, but will not fail the execution.
+            </td>
+          </tr>
+
+          <tr>
+            <td><code>warning</code></td>
+            <td>
+              This severity behaves exactly the same as <code>info</code> in Checkstyle.
+              It can be used to let user bring more attention to reviewers of such violations.
+              If executing Checkstyle through Ant,
+              <a href="anttask.html#Parameters"><code>maxWarnings</code></a> can be used to pass
+              the execution until a certain number of warnings are found.
+            </td>
+          </tr>
+
+          <tr>
+            <td><code>error</code></td>
+            <td>
+              If a violation occurs, modules with this severity are displayed as error and should
+              be considered as a failure.
+              Depending on how Checkstyle is executed, there may be some additional settings which
+              can be set to not fail the execution or until a certain number of errors occur.
```

Comment:
I think we should mention that CLI and Ant will fail. We do not know about any others behavior of plugins

Reply:
>be considered as a failure.

Hmm, I still think it not good wording. I will think on it tomorrow

Reply:
> I think we should mention that CLI and Ant will fail. 

Done.

> We do not know about any others behavior of plugins

Agreed, but we should document what we believe the expected behavior is. Plugins can always change it for their needs, but our view of the severity will not change. Ex: We feel errors should fail the execution even if another plugin says at least X violations must occur first.

Reply:
audit listeners does not make decision on failure/(error-code) of execution
https://github.com/checkstyle/checkstyle/blob/a8f065738de6c4ff917a2ee57521f6cf8718f2ba/src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

decusion on when to fail in 
wrapper of Checkstyle: CLI, ANT, any plugin ....

example:
https://github.com/checkstyle/checkstyle/blob/b1c44756f64cb1009e9648cf107e7e467c27197c/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L130 

lets be explicit. to make it clear that each wrapper works differently and better to refer to its documentation to know when execution fail.

Reply:
may be we should make two more columns: CLI and ANT  and describe what such wrapers are doing with events severity.

CLI does not fail on WARN, and has no arguments to contomize failure behavior.
ANT is more advanced and has limits on error and warn severity.

Reply:
This is why I didn't want to get too depth into specifics with my original wording. I was giving an overview and then saying actual behavior depends on execution method. We could link to those sections so users can see more details over there. I worry if we put all details here, we will forget to update it later when we add new functionality.